modules/nodelist.php

```diff
@@ -58,7 +58,14 @@
 	$ng = $_GET['ng'];
 $SESSION->save('nlng', $ng);
 
-$nodelist = $LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng);
+$page = (!isset($_GET['page']) ? 1 : intval($_GET['page']));
+$per_page = intval(ConfigHelper::getConfig('phpui.nodelist_pagelimit', $listdata['total']));
+$offset = ($page - 1) * $per_page;
+$total = intval($LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng, null, null, true));
+
+$nodelist = $LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng, $per_page, $offset);
```

Comment:
Ale to nadal nie zmienia faktu, że nie chcemy liczyć zwróconych rekordów,
bo mamy limit, tylko wszystkie spełniające warunki wyszukiwania
24 lis 2015 15:39 "Tomasz Chiliński" <notifications@github.com> napisał(a):

> In modules/nodelist.php
> <https://github.com/lmsgit/lms/pull/449#discussion_r45742603>:
>
> > @@ -58,7 +58,14 @@
> >  	$ng = $_GET['ng'];
> >  $SESSION->save('nlng', $ng);
> >
> > -$nodelist = $LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng);
> > +$page = (!isset($_GET['page']) ? 1 : intval($_GET['page']));
> > +$per_page = intval(ConfigHelper::getConfig('phpui.nodelist_pagelimit', $listdata['total']));
> > +$offset = ($page - 1) * $per_page;
> > +$total = intval($LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng, null, null, true));
> > +
> > +$nodelist = $LMS->GetNodeList($o, NULL, NULL, $n, $s, $g, $ng, $per_page, $offset);
>
> To trzeba byłoby do sterowników mysql, mysqli i postgres w LMS dodać
> odpowiednią metodę, która zasłoniłaby przez twórcą zapytań SQL w LMS
> specyfikę bazy danych. Ktoś chętny to drobienia tego? ;-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lmsgit/lms/pull/449/files#r45742603>.
>